src/middlewares/authentication.ts

```diff
@@ -0,0 +1,65 @@
+import { Request, Response, NextFunction } from 'express';
+import jsonwebtoken from 'jsonwebtoken';
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+import { User } from '../models/user';
+import { UserRoleEnum } from '../utils/common/enum';
+interface IJWTPayload {
+  _id: string;
+}
+
+class HandleAuthentication {
+  public async userAuthentication(
+    req: Request,
+    _res: Response,
+    next: NextFunction
+  ): Promise<Response | void> {
+    const authToken = req.header('Authorization')?.replace('Bearer ', '');
+    if (!authToken) {
+      next();
+      return;
+    }
+    const decoded = jsonwebtoken.verify(
+      authToken as string,
+      process.env.JWT_SECRET_KEY as string
+    );
+    const userId = decoded as IJWTPayload;
+
+    const user = await User.findOne({
+      _id: userId._id
+    });
+
+    if (user && user.enabled === true) {
+      req.authToken = authToken;
+      req.user = user;
+    }
+    next();
```

Comment:
Invert

```suggestion
    if (authToken) {
          const decoded = jsonwebtoken.verify(
          authToken as string,
          process.env.JWT_SECRET_KEY as string
        );
        const userId = decoded as IJWTPayload;
    
        const user = await User.findOne({
          _id: userId._id
        });
    
        if (user && user.enabled === true) {
          req.authToken = authToken;
          req.user = user;
        }
    }
    
    next();
```

Then you can refactor a bit

```suggestion
    if (authToken) {
        const payload = decodeJWTToken(authToken as string);
 
        const user = await User.findOne({
          _id: payload._id
        });
    
        if (user && user.enabled === true) {
          req.authToken = authToken;
          req.user = user;
        }
    }
    
    next();
```